app/controllers/users/registrations_controller.rb

```diff
@@ -29,10 +29,20 @@ def destroy
     end
   end
 
+  def password
+    redirect_to root_path if current_user.nil? 
+  end
+
   protected
 
-  def update_resource(resource, params)
+  def update_resource(resource, params) 
+    if params[:password_confirmation].nil? && resource.errors.size == 0
+      flash.now[:notice] = 'User succesfully updated!'
```

Comment:
Keep the "green" flash notice and delete the "red" flash notice (usually means error).